### PR TITLE
Feature/log viewer

### PR DESCRIPTION
**Pull request by 0x6368**
**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature

**What is the current behavior?** (You can also link to an open issue here)
Opening the logs, downloads the whole log file (even if it it multiple megabytes in size)

**What is the new behavior (if this is a feature change)? If possible add a screenshot.**
Opening the logs opens a log viewer, that only loads the last lines of the logs. You can scroll through the logs by using the arrow keys or pgup/pgdown keys
image

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
A little bit? The option to download logs has moved to the log viewer instead of the report overview. The report overview will only link to the new log viewer

**Other information**:
---

### CODE REVIEW

**Comment by 0x6368** [comment]
I'll fix the linter issues in the coming days

**Comment by 0x6368** [comment]
I tried installing watchdog, but somehow it cannot lock the pipfile:

```
ubuntu@ubuntu-virtual-machine:~/embark$ pipenv install watchdog
Loading .env environment variables...
Installing watchdog...
Resolving watchdog...
Added watchdog to Pipfile's [packages] ...
✔ Installation Succeeded
Pipfile.lock (8a1083) out of date, updating to (78152a)...
Locking [packages] dependencies...
Building requirements...
Resolving dependencies...
✘ Locking Failed!
⠼ Locking...False
ERROR:pip.subprocessor:[present-rich] Getting requirements to build wheel exited with 1
[ResolutionFailure]:   File "/usr/local/lib/python3.10/dist-packages/pipenv/resolver.py", line 645, in _main
[ResolutionFailure]:       resolve_packages(
[ResolutionFailure]:   File "/usr/local/lib/python3.10/dist-packages/pipenv/resolver.py", line 612, in resolve_packages
[ResolutionFailure]:       results, resolver = resolve(
[ResolutionFailure]:   File "/usr/local/lib/python3.10/dist-packages/pipenv/resolver.py", line 592, in resolve
[ResolutionFailure]:       return resolve_deps(
[ResolutionFailure]:   File "/usr/local/lib/python3.10/dist-packages/pipenv/utils/resolver.py", line 897, in resolve_deps
[ResolutionFailure]:       results, hashes, internal_resolver = actually_resolve_deps(
[ResolutionFailure]:   File "/usr/local/lib/python3.10/dist-packages/pipenv/utils/resolver.py", line 670, in actually_resolve_deps
[ResolutionFailure]:       resolver.resolve()
[ResolutionFailure]:   File "/usr/local/lib/python3.10/dist-packages/pipenv/utils/resolver.py", line 447, in resolve
[ResolutionFailure]:       raise ResolutionFailure(message=str(e))
[pipenv.exceptions.ResolutionFailure]: Warning: Your dependencies could not be resolved. You likely have a mismatch in your sub-dependencies.
  You can use $ pipenv run pip install <requirement_name> to bypass this mechanism, then run $ pipenv graph to inspect the versions actually installed in the virtualenv.
  Hint: try $ pipenv lock --pre if it is a pre-release dependency.
ERROR: Getting requirements to build wheel exited with 1
```

**Comment by BenediktMKuehne** [comment]
Oh that's always fun when that happens.😬

One of the packages has a dependency conflict with your new one.
Quite possibly the wsgi.
You could try forcing a newer version for the dependency

**Comment by 0x6368** [comment]
Looks like this was an issue with mysqlclient. This helped my install watchdog: https://forum.codewithmosh.com/t/pipenv-install-mysql-client/20899/4

In my case, I had to change the environment variables a little:

```bash
export MYSQLCLIENT_LDFLAGS='-L/usr/mysql/lib -lmysqlclient -lssl -lcrypto -lresolv'
export MYSQLCLIENT_CFLAGS='-I/usr/include/mysql/'
```

File: installer.sh [hunk 1]
```diff
@@ -344,6 +344,8 @@ install_embark_default(){
     wget -O ./embark/static/external/scripts/bootstrap.js https://cdn.jsdelivr.net/npm/bootstrap@5.2.3/dist/js/bootstrap.bundle.min.js
     wget -O ./embark/static/external/scripts/datatable.js https://cdn.datatables.net/v/bs5/dt-1.11.2/datatables.min.js
     wget -O ./embark/static/external/scripts/charts.js https://cdn.jsdelivr.net/npm/chart.js@3.5.1/dist/chart.min.js
+    wget -O ./embark/static/external/scripts/base64.js https://cdn.jsdelivr.net/npm/js-base64@3.7.5/+esm
```
Comment (BenediktMKuehne) [inline installer.sh:347]: Can we add these two lines to the developer install function too, please
line 424 for example

**Comment by BenediktMKuehne** [comment]
will be testing the changes now

**Review by BenediktMKuehne — CHANGES_REQUESTED** [review]

**Comment by BenediktMKuehne** [comment]
Result is looking great 👍 
(sorry it took so long, I forgot to send/finalize the review)

File: installer.sh (hunk 1 above)
Comment (0x6368) [reply installer.sh:347]: Done

**Review by BenediktMKuehne — APPROVED** [review]

**Comment by BenediktMKuehne** [comment]
Nice work!

**Pull request merged**